contracts/base/LoanTermsConsensus.sol

```diff
@@ -22,17 +27,75 @@ import "../interfaces/LoanTermsConsensusInterface.sol";
 
     @author develop@teller.finance
  */
-contract LoanTermsConsensus is LoanTermsConsensusInterface, Consensus {
-    /* Mappings */
+contract LoanTermsConsensus is LoanTermsConsensusInterface, OwnerSignersRole {
+    using SafeMath for uint256;
+    using NumbersList for NumbersList.Values;
+    using NumbersLib for uint256;
+
+    /** Address of the loanManager */
+    address public loanManagerAddress;
 
+    /* Mappings */
     /**
         This mapping identify the last request timestamp for a given borrower address.
 
             Example:    address(0x123...789) = timestamp(now)
 
         It is used as rate limit per borrower address.
      */
-    mapping(address => uint256) public borrowerToLastLoanTermRequest;
+    mapping(address => uint256) public borrowerRequestTimes;
+
+    /**
+        @notice It tracks each request nonce value used by the borrower.
+     */
+    mapping(address => uint256) public borrowerNonces;
```

Comment:
we can get rid of these mappings and use the values from `LoanStorage` 
* `borrowerRequestTimes[borrower]` becomes `loans[borrowerLoans[borrower][borrowerLoans[borrower].length - 1].loanStartTime`
* `borrowerNonces[borrower]` becomes `borrowerLoans[borrower].length`